### PR TITLE
Vocabulary component fixes

### PR DESCRIPTION
**Pull request by pmaria**
* The vocabulary appearance shows all properties on every class. This is due to the wrong property list attribute being referenced while building the list. 0ea66c6 fixes this.
* The vocabulary appearance determines class - property relationships based on SHACL alone. As SHACL is not always used, it makes sense to also support RDFS and OWL constructs to determine these relations. This PR makes a start by supporting broadly used rdfs:domain and rdf:range constructs. It also works when both RDFS constructs and SHACL constructs are used.

### CODE REVIEW

File: src/components/ClassList.tsx [hunk 1]
```diff
@@ -32,20 +33,26 @@ const findAncestorClassIris = (classIri: Term, store: Store): Term[] => {
 
 const findPropertyIris = (classIri: Term, store: Store): Term[] => {
   const shapeIri = store.findSubjects(namedNode(SHACL + 'targetClass'), classIri)[0];
+  const properties = store.findSubjects(namedNode(RDFS + 'domain'), classIri);
 
-  if (!shapeIri) {
+  if (!shapeIri && properties.length == 0) {
```
Comment (joostfarla) [inline src/components/ClassList.tsx]: `===` ipv `==`

File: src/components/ClassList.tsx [hunk 2]
```diff
@@ -32,20 +33,26 @@ const findAncestorClassIris = (classIri: Term, store: Store): Term[] => {
...
-  return store
-    .findObjects(shapeIri, namedNode(SHACL + 'property'))
-    .reduce(
-      (acc: Term[], propertyShapeIri: Term) => [
-        ...acc,
-        ...store.findObjects(propertyShapeIri, namedNode(SHACL + 'path')),
-      ],
-      [],
-    );
+  var shProperties : Term[] = [];
```
Comment (joostfarla) [inline src/components/ClassList.tsx]: Liever in 1 statement ivm immutability:
```js
const propertyIris = store
  .findSubjects(namedNode(RDFS + 'domain'), classIri)
  .concat(shapeIri === undefined ? [] :
    store
      .findObjects(...)
      .reduce(...));
```

File: src/components/ClassList.tsx [hunk 3]
```diff
@@ -32,20 +33,26 @@ const findAncestorClassIris = (classIri: Term, store: Store): Term[] => {
 
 const findPropertyIris = (classIri: Term, store: Store): Term[] => {
   const shapeIri = store.findSubjects(namedNode(SHACL + 'targetClass'), classIri)[0];
+  const properties = store.findSubjects(namedNode(RDFS + 'domain'), classIri);
```
Comment (joostfarla) [inline src/components/ClassList.tsx]: Suffixen met `Iris` ivm naming consistentie

File: src/components/PropertyList.tsx [hunk 4]
```diff
@@ -59,6 +68,25 @@ const findRelatedClassIri = (propertyShapeIris: Term[], store: Store): Term | un
...
+        return acc;
+      }
+
+      // Filter out XSD datatypes
+      // TODO: this is quite crude, and doesn't filter out other datatypes.
+      if(related.value.indexOf(XSD) !== 0) {
+        return related;
+      }
+    },
+    undefined,
```
Comment (joostfarla) [inline src/components/PropertyList.tsx]: Initial value voor accumulator lijkt overbodig? `undefined` is default behaviour.

File: package.json [hunk 5]
```diff
@@ -18,9 +18,11 @@
   },
   "dependencies": {
     "@rdfjs/data-model": "^1.1.1",
+    "@types/lodash": "^4.14.123",
     "es6-promise": "^4.2.5",
     "i18next": "^12.1.0",
     "i18next-browser-languagedetector": "^2.2.4",
+    "lodash": "^4.17.11",
```
Comment (joostfarla) [inline package.json]: Graag Ramda functies gebruiken ipv `lodash`

File: package.json [hunk 6]
```diff
@@ -51,6 +53,7 @@
     "@types/react-router-dom": "^4.3.1",
     "@types/react-treeview": "^0.4.2",
     "@types/sinon": "^7.0.5",
+    "acorn": "^6.1.1",
```
Comment (joostfarla) [inline package.json]: Wordt niet gebruikt.

File: src/components/ClassList.tsx [hunk 7]
```diff
@@ -32,20 +33,26 @@ const findAncestorClassIris = (classIri: Term, store: Store): Term[] => {
...
+      .reduce(
+        (acc: Term[], propertyShapeIri: Term) => [
+          ...acc,
+          ...store.findObjects(propertyShapeIri, namedNode(SHACL + 'path')),
+        ],
+        [],
+      );
+  }
+  const combined = shProperties.concat(properties);
+  return uniqWith(combined, (term, other) => term.equals(other));
```
Comment (joostfarla) [inline src/components/ClassList.tsx]: Newline boven `return` statement, als deze volgt op een multi-line statement.

File: src/components/PropertyList.tsx [hunk 8]
```diff
@@ -59,6 +68,25 @@ const findRelatedClassIri = (propertyShapeIris: Term[], store: Store): Term | un
...
+const findRdfsRangeRelatedClassIri = (propertyIri: Term, store: Store): Term | undefined =>
+  store.findObjects(propertyIri, namedNode(RDFS + 'range')).reduce(
+    (acc: Term | undefined, related) => {
+      if (acc !== undefined) {
+        return acc;
+      }
+
+      // Filter out XSD datatypes
+      // TODO: this is quite crude, and doesn't filter out other datatypes.
+      if(related.value.indexOf(XSD) !== 0) {
```
Comment (joostfarla) [inline src/components/PropertyList.tsx]: Spatie na `if`

File: src/components/PropertyList.tsx (hunk 4 above)
Comment (pmaria) [reply src/components/PropertyList.tsx]: Als ik het verwijder gaat de compiler zeuren...
`Type 'undefined' is not assignable to type 'Term'`. Uitvoeren leidt tot
`Reduce of empty array with no initial value`.

Comment (joostfarla) [reply src/components/PropertyList.tsx]: Laat deze dan maar zo staan.

**Pull request merged**